pkg/scheduler/objects/application.go

```diff
@@ -1264,6 +1264,9 @@ func (sa *Application) AddAllocation(info *Allocation) {
 // Add the Allocation to the application
 // No locking must be called while holding the lock
 func (sa *Application) addAllocationInternal(info *Allocation) {
+	if sa.queue != nil && sa.queue.maxRunningApps != 0 && sa.stateMachine.Is(Accepted.String()) && sa.queue.maxRunningApps <= sa.queue.runningApps {
+		return
```

Comment:
I don't think we can just return.

The application cannot continue its transition because maxRunningApps is exceeded.

Therefore, it has to be saved to a slice along with the current timestamp so that when `queue.runningApps < queue.maxRunningApps` we can manually trigger the state transition.

The timestamp is needed because we want to trigger the app which has been waiting for the longest amount of time.

Reply:
As @wilfred-s mentioned in the JIRA, we have to have a `CanRun()` method in the queue, something like:

```
func (sq *Queue) CanRun() {
	sq.RLock()
	defer sq.RUnlock()
	
	ok := sq.runningApps < sq.maxRunningApps
	if sq.parent != nil {
		return ok && sq.parent.CanRun()
	} else {
		return ok
	}
}
```